### ISSUE TITLE
indentation indicator tones are delayed when using Firefox

### ISSUE THREAD

**Issue by mehgcap** [nvaccess/nvda#10605]
### Steps to reproduce:
Using Firefox, open any webpage. Be sure you are in Browse Mode, and that the page has some indentation on it. Also be sure to enable indentation indication and set it to tones. I haven't tested using speech instead of tones.

### Actual behavior:
Tones are delayed. When you arrow to a line that has a different indentation level, the tone plays well after speech starts to speak the line.
### Expected behavior:
As in text editors, the tone should play as or before speech begins to speak the line of text.
#### NVDA installed/portable/running from source:
portable
#### NVDA version:
2019.3 beta 1
#### Windows version:
10, 1903
#### Name and version of other software in use when reproducing the issue:
Firefox 71.0
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2019.2.1 and earlier do not have this problem

**Comment by Adriani90** [comment]
cc: @jcsteh, @leonardder

**Comment by jcsteh** [comment]
My guess is that what you're noticing is that the tone now occurs at the start of the actual text. So, if the text is "test" and is a visited link, you'll hear the tone after the words "visited link"; i.e. the tone will play with "test". While this is certainly different, I don't see this as a bug. If you were reporting indentation with speech, you'd hear "link 2 space test", not "2 space link test". If anything, this is more consistent.

**Comment by XLTechie** [comment]
On Tue, 10 Dec 2019, James Teh wrote:

> My guess is that what you're noticing is that the tone now occurs at the start of the actual text. So, if the text is "test" and is a visited link, you'll
> hear the tone after the words "visited link"; i.e. the tone will play with "test". While this is certainly different, I don't see this as a bug. If you were
> reporting indentation with speech, you'd hear "link 2 space test", not "2 space link test". If anything, this is more consistent.

Consistent with what? Not with fact, surely.

The two spaces are not part of the visited link. But that makes it seem that 
they are.

The two spaces come before the text which is the visited link.
So they should not be named as being part of the visited link.
That is changing the nature of the link, by making the indentation part of it.

So if a blind user should then want to tell a sighted one about that link, he 
will describe a link that has text preceded by X number of spaces in the 
highlighted clickable part of the link, when in fact it is only the text that is 
the link.

An absurd notion for purposes of example, but if we're going to start changing 
the order of screen elements, we better consider carefully what the binding 
precedence should be.

In my mind, if text is a link, only the text that is a link should be called 
"link". If text is blue, only the text that is blue should be called "blue", not 
the text and the spaces around it which probably aren't blue.

If we're trying to give a blind user a good idea of how the screen looks, or how 
some elements look, saying "visited link" before the indentation is quite 
inaccurate.

**Comment by mehgcap** [comment]
Thanks, @JCSteh. This is exactly what I'm experiencing, and now that I know what I'm hearing, it makes sense. I'm not sure how I feel about it yet, but I suspect it's just something new. In two weeks, I'll be so used to it I won't even think about it, and going back to the previous way will feel as odd as this new way does today.

As to @XLTechie's point, I see what you're saying, but I don't think that's quite what's happening. NVDA is simply delaying the indent tone until after the HTML details have been dealt with. Information like "main landmark heading level 1" is spoken, then the indent plays. The tone has to play sometime, after all, and the only difference here is that it waits until NVDA is starting to speak the text. Previously, the tone would play immediately, which offers no more or less indication of where the indent is in relation to the link than you get with the new method. Basically, the new delay doesn't change whether the user considers spaces to be part of the linked text. At least, that's my understanding.

**Comment by jcsteh** [comment]
> Consistent with what? Not with fact, surely. The two spaces are not part of the visited link. But that makes it seem that they are.

Why do you assume the two spaces aren't part of the link? They can be and often are. Weird, yes, but does it happen, absolutely. Consider this example:

`data:text/html,<p>Next line is link with spaces before it</p>&nbsp;&nbsp;<span role="link">test</span><p>Next line is link containing spaces</p><span role="link">&nbsp;&nbsp;test`

You'll see that in the first link, NVDA reports the indentation before the link, because that's where it is. In contrast, for the second link, NVDA reports the indentation after the reporting of "link" because the spaces are part of the link.

**Comment by Adriani90** [comment]
Just a thought from my side, could this not confuse users who might think that the indentation is to the right edge of the screen and not from left edge? I mean if the tone is played after the link is spoken, the could create this impression.

**Comment by jcsteh** [comment]
It's played after the word link is spoken, not the whole link. So, if the
spaces are part of the link and the rest of the text in the link is "test",
you might hear "link", followed by the beep and the word "text" in parallel.

**Comment by Adriani90** [comment]
Ok I should have read more carefully the comments above :-). Now I understand. In this case I am closing this issue as works for me.
